src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -336,11 +336,34 @@ private async ValueTask<ProtocolToken> ReceiveBlobAsync<TIOAdapter>(TIOAdapter a
             {
                 await FillHandshakeBufferAsync(adapter, frameSize).ConfigureAwait(false);
             }
+            // At this point, we have at least one TLS frame.
 
-            ProtocolToken token = _context.NextMessage(_handshakeBuffer.ActiveReadOnlySpan.Slice(0, frameSize));
+            return ProcessBlob(frameSize);
+        }
+
+        // Calls crypto on received data. No IO inside.
+        private ProtocolToken ProcessBlob(int frameSize)
+        {
+            int chunkSize = frameSize;
+
+            ReadOnlySpan<byte> availableData = _handshakeBuffer.ActiveReadOnlySpan;
             _handshakeBuffer.Discard(frameSize);
```

Comment:
I'm not clear what's happening here.  We're getting a span and then calling Discard... is that not invalidating some of the data in the span?

Reply:
No, it does not AFAIK. The name may be confusing but it only increments starting index and that impacts ActiveLength and ActiveSpan. I also added a comment to make it more clear. I could rewrite it with Slices but that would be more logic IMHO instead of depending on ArrayBuffer itself.

Reply:
Ok